backend/aurcache/src/builder/build.rs

```diff
@@ -266,7 +266,8 @@ impl Builder {
 
     /// move built files from build container to host and add them to the repo
     async fn move_and_add_pkgs(&self, host_build_path: PathBuf) -> anyhow::Result<()> {
-        let archive_paths = fs::read_dir(host_build_path.clone())?.collect::<Vec<_>>();
+        let archive_paths: Vec<_> =
+            fs::read_dir(host_build_path.clone())?.collect::<Result<_, _>>()?;
```

Comment:
I think it's safer to return here if there's a I/O error on one of the entries, rather than further down after removing the old files?